tests/tests_raid_volume_cleanup.yml

```diff
@@ -0,0 +1,105 @@
+---
+- hosts: all
+  become: true
+  vars:
+    storage_safe_mode: false
+    storage_use_partitions: true
+    mount_location1: '/opt/test1'
+    mount_location2: '/opt/test2'
+    volume1_size: '5g'
+    volume2_size: '4g'
+
+  tasks:
+    - include_role:
+        name: linux-system-roles.storage
+
+    - name: Mark tasks to be skipped
+      set_fact:
+        storage_skip_checks:
+          - blivet_available
+          - packages_installed
+          - service_facts
+
+    - include_tasks: get_unused_disk.yml
+      vars:
+        max_return: 3
+        disks_needed: 3
+
+    - name: Create two LVM logical volumes under volume group 'foo'
+      include_role:
+        name: linux-system-roles.storage
+      vars:
+        storage_pools:
+          - name: foo
+            disks: "{{ unused_disks }}"
+            volumes:
+              - name: test1
+                size: "{{ volume1_size }}"
+                mount_point: "{{ mount_location1 }}"
+              - name: test2
+                size: "{{ volume2_size }}"
+                mount_point: "{{ mount_location2 }}"
+
+    - name: Enable safe mode
+      set_fact:
+        storage_safe_mode: true
+
+    - name: Try to overwrite existing device with raid volume and safe mode on (expect failure)
+      block:
+        - name: Create a RAID0 device mounted on "{{ mount_location1 }}"
```

Comment:
For checking that the role raises an error, and that the error message is the one that you expect, please consider using verify-role-failed.yml as in https://github.com/linux-system-roles/storage/blob/main/tests/tests_misc.yml#L66
One of the issues we have had in the past with testing for storage role failures was that the role would fail in the wrong way i.e. because the inputs were not correct, or there was some other blivet error, but the test would report that the role failed correctly because it was not checking for the error message - using verify-role-failed.yml makes it easy to verify that the role failed *AND* that the error message is the correct one for the failure condition.